test/core/api/client/http_client.hpp

```diff
@@ -48,14 +48,13 @@ namespace test {
     /**
      * @param context reference to io context instance
      */
-    explicit ApiClient(Context &context)
-        : stream_(context) {}
+    explicit HttpClient(Context &context) : stream_(context) {}
 
-    ApiClient(const ApiClient &other) = delete;
-    ApiClient &operator=(const ApiClient &other) = delete;
-    ApiClient(ApiClient &&other) noexcept = delete;
-    ApiClient &operator=(ApiClient &&other) noexcept = delete;
-    ~ApiClient();
+    HttpClient(const HttpClient &other) = delete;
```

Comment:
Maybe better inherit from boost::noncopyable?

Reply:
Probably for this case, but in general not. boost::noncopyable was useful before or early  C++11 compilers, but right now it's obsolete.
Look at https://stackoverflow.com/questions/7823990/what-are-the-advantages-of-boostnoncopyable